### DISCUSSION TITLE
Join Without FK Constraint

### DISCUSSION THREAD

**Discussion by achempak-polymer** [aarondl/sqlboiler#1194] [Q&A] [answered]
Is there a way to perform joins on tables that don't have explicit foreign keys in the database? Our database doesn't make use of foreign keys. When I try to perform joins using sqlboiler, it throws an error 
```
"Failed to get foos models: failed to assign all query results to Foo slice: bind failed to execute query: ERROR: relation \"bars\" does not exist
```
The error is expected, but is there a (relatively simple) way to work around this? If not, I would either have to add foreign keys to the db (which I want to avoid), or use a different query builder/ORM. For example, is there some way to give a list of virtual foreign keys to sqlboiler?

Thanks! 😄

**Comment by stephenafamo** [comment, marked as answer]
I don't understand the question. When building a query, just like with regular SQL, you can do joins to other tables. e.g.:

```go
qm.InnerJoin()
qm.LeftJoin()
// e.t.c.
```

This has nothing to do with foreign keys
